derive/src/decode.rs

```diff
@@ -87,9 +87,9 @@ fn create_decode_expr(field: &Field, name: &proc_macro2::TokenStream, input: &To
 		quote_spanned! { field.span() =>
 			{
 				let res = <<#field_type as _parity_codec::HasCompact>::Type as _parity_codec::Decode>::decode(#input);
-				let fname = stringify!(#name).replace(" ", "");
+				let msg = stringify!(Error decoding field #name);
```

Comment:
Please put the msg above the if and do something like `let msg = format!("Error decoding field: {}", name);` and then just use `#msg` in `quote!`